linalg/solvers.hpp

```diff
@@ -1125,6 +1125,40 @@ class ProductSolver : public Solver
    virtual void SetOperator(const Operator &op) { }
 };
 
+/// Solver wrapper which orthogonalizes the input and output vector
+/**
+ * OrthoSolver wraps an existing Operator and orthogonalizes the input vector
+ * before passing it to the Mult method of the Operator. This is a convenience
+ * implementation to handle e.g. a Poisson problem with pure Neumann boundary
+ * conditions, where this procedure removes the Nullspace.
+ */
+class OrthoSolver : public Solver
+{
+private:
+#ifdef MFEM_USE_MPI
+   MPI_Comm mycomm;
+#endif
+
+   const bool parallel;
+
+public:
+   OrthoSolver();
+#ifdef MFEM_USE_MPI
+   OrthoSolver(MPI_Comm mycomm_);
+#endif
+
+   virtual void SetOperator(const Operator &op);
```

Comment:
This should probably be replaced with something named `SetSolver` or similar, since if this is used in the context of e.g. `CGSolver`, `SetOperator` will be called with the *forward operator* rather than the preconditioner, which will be incorrect here.

`SetOperator` should just propagate the forward operator to the preconditioner.

Reply:
done, but @jandrej should check if Navier still works as expected.

Reply:
No change in output for Navier.